src/components/token/TokenBalances.js

```diff
@@ -5,372 +5,367 @@
-     */
-    this.state = {
-      tokenId,
-      tokenBalances: [],
-      hasAfter: false,
-      hasBefore: false,
-      sortBy,
-      order,
-      page: 1,
-      pageSearchAfter: [],
-      loading: true,
-      isSearchLoading: false,
-      calculatingPage: false,
-      error: false,
-      tokenBalanceInformationError: false,
-      maintenanceMode: this.props.maintenanceMode,
-      transactionsCount: 0,
-      addressesCount: 0,
-      tokensApiError: false,
-    };
-  }
+    const queryParams = pagination.current.obtainQueryParams();
 
-  componentDidMount = async () => {
-    if (this.state.maintenanceMode) {
-      return;
-    }
+    const querySortBy = get(queryParams, 'sortBy', 'total');
+    const queryOrder = get(queryParams, 'order', 'desc');
+    const queryTokenId = get(queryParams, 'token', hathorLibConstants.NATIVE_TOKEN_UID);
 
-    if (this.state.tokenId === hathorLibConstants.NATIVE_TOKEN_UID) {
+    setSortBy(querySortBy);
+    setOrder(queryOrder);
+    setTokenId(queryTokenId);
+
+    if (queryTokenId === hathorLibConstants.NATIVE_TOKEN_UID) {
       // If we had a custom token as queryParam
       // then we will perform the search after the token
       // is found in the elastic search
       // otherwise we just perform the search for HTR to show the default screen
-      await this.performSearch();
+      setIsSearchLoading(true);
 
       // Since we did not search for the HTR token (it is the default), we need to fetch
       // it to retrieve the transactions count
-      await this.fetchHTRTransactionCount();
-
-      this.setState({
-        loading: false,
-      });
+      fetchHTRTransactionCount().catch(e =>
+        console.error('Error on initial fetchHTRTransactionCount', e)
+      );
     }
-  };
+  }, [fetchHTRTransactionCount]);
 
   /**
    * Call explorer-service to get list of token balances for a given tokenId
    *
+   * @param queryTokenId
+   * @param querySortBy
+   * @param queryOrder
    * @param {*} searchAfter Parameter needed by ElasticSearch for pagination purposes
    * @returns tokens
    */
-  getTokenBalances = async searchAfter => {
-    const tokenBalancesRequest = await tokensApi.getBalances(
-      this.state.tokenId,
-      this.state.sortBy,
-      this.state.order,
-      searchAfter
-    );
-    this.setState({
-      error: get(tokenBalancesRequest, 'error', false),
-    });
-    return get(tokenBalancesRequest, 'data', { hits: [], has_next: false });
-  };
+  const getTokenBalances = useCallback(
+    async (queryTokenId, querySortBy, queryOrder, searchAfter) => {
+      const tokenBalancesResponse = await tokensApi.getBalances(
+        queryTokenId,
+        querySortBy,
+        queryOrder,
+        searchAfter
+      );
+      setError(tokenBalancesResponse.error || false);
+      return tokenBalancesResponse.data || { hits: [], has_next: false };
+    },
+    []
+  );
+
+  const loadTokenBalanceInformation = useCallback(async queryTokenId => {
```

Comment:
The way I structured this component, the `useEffect()` on line 168 is the one responsible for obtaining the most up-to-date `queryTokenId` and pass it as a parameter for this callback function.

Since the `loadTokenBalanceInformation` function will always receive all needed parameters directly through its calls, it never needs to be re-rendered because of any state changes, so its dependencies are empty.

I considered the possibility of removing it from inside the component altogether, but it needs to update the `tokenBalanceInformationError` state, so I left it inside the component. A more profound refactor could have done this, but I preferred not changing too much the logic of the component for now.

Note: the `set` functions are actually stable and need not be informed as dependencies for callbacks and effects ( see [relevant docs](https://react.dev/learn/synchronizing-with-effects#why-was-the-ref-omitted-from-the-dependency-array) )